src/core/DatetimeClasses.ts

```diff
@@ -0,0 +1,367 @@
+    }
+
+    /**
+     * Returns the string representation of this local date.
+     * @returns A string in the form yyyy:mm:dd. Values are zero padded from left
+     */
+    toString(): string {
+        const sign = this.year < 0 ? '-' : '';
+        const paddedYear = Math.abs(this.year).toString().padStart(4, '0');
+        const month = this.month.toString().padStart(2, '0');
+        const date = this.date.toString().padStart(2, '0');
+        return `${sign}${paddedYear}-${month}-${date}`;
+    }
+}
+
+/**
+ * ### Local datetime object
+ * * Represents date and time without timezone.
+ */
+export class HzLocalDateTime {
+    /**
+     * @throws RangeError if arguments are invalid
+     */
+    constructor(
+        readonly hzLocalDate: HzLocalDate,
+        readonly hzLocalTime: HzLocalTime
+    ) {
+        if (!(hzLocalDate instanceof HzLocalDate)) {
+            throw new RangeError('Invalid local date.');
+        }
+        if (!(hzLocalTime instanceof HzLocalTime)) {
+            throw new RangeError('Invalid local time.');
+        }
+    }
+
+    /**
+     * Constructs HzLocalDateTime from ISO 8601 string.
+     * @param isoString Must not include timezone information. The string format is yyyy-mm-ss(T|t)HH:mm:ss.SSS. The last SSS
+     * part represents nanoseconds and can be omitted.
+     * @throws RangeError if ISO string is invalid or any of the values in ISO string is invalid
+     */
+    static fromString(isoString: string): HzLocalDateTime {
+        if (typeof isoString !== 'string') {
+            throw new RangeError('String expected.');
+        }
+        const split = isoString.split(/[Tt]/);
+        if (split.length !== 2) {
+            throw new RangeError('Invalid format. Expected a string in the form yyyy-mm-ss(T|t)HH:mm:ss.SSS');
+        }
+        return new HzLocalDateTime(HzLocalDate.fromString(split[0]), HzLocalTime.fromString(split[1]));
+    }
+
+    /**
+     * Returns the string representation of this local datetime.
+     * @returns A string in the form yyyy:mm:ddTHH:mm:ss.SSS. Values are zero padded from left. If nano second value
+     * is zero, second decimal is not include in the returned string
+     */
+    toString(): string {
+        return `${this.hzLocalDate.toString()}T${this.hzLocalTime.toString()}`;
+    }
+}
+
+/**
+ * ### Offset datetime object
+ * * Represents date and time with timezone.
+ * * Timezone is specified with offset from UTC in seconds. This offset can be negative or positive.
+ * * This class internally stores a {@link HzLocalDateTime} and offset number.
+ */
+export class HzOffsetDateTime {
+
+    hzLocalDateTime: HzLocalDateTime;
+    private static readonly timezoneRegex = /([Zz]|[+-]\d\d:\d\d)/;
+
+    /**
+     * @param date Must be a valid date. So `date.getTime()` should be not NaN
+     * @param offsetSeconds Must be between -64800-64800 (-+18:00)
+     */
+    constructor(date: Date, readonly offsetSeconds: number) {
```

Comment:
I think if we are going to accept Date objects in a constructor, why not we do it in the constructor of the `HzLocalDateTime`? I feel like it is more or less similar to Date. Personally, I wouldn't put it into the constructor, but put a factory method like `HzLocalDateTime.fromDate`. What do you think about it?

Reply:
I agree. In order to have similar constructors, I will move fromDate to static constructor and make constructor accept HzLocalDateTime. Do you think I should add fromDate to HzLocalDatetime as well?

Reply:
I felt like we should only add it to `HzLocalDatetime`. But if we do that, we will ignore the timezone information of the Date object. So, I leave the decision to you

I think if we add `HzOffsetDateTime.fromDate`, it should take the offset information from date but I am not sure about it.

Reply:
Date objects does not store timezone information. Added them you can check.